### PR TITLE
Adds a Material Stack manipulation apparatus for Engineer Cyborgs, replaces the R-Glass tool

### PR DESCRIPTION
**Pull request by zxaber**
## About The Pull Request
- Adds a new apparatus for Engineer Cyborgs that can pick up and manipulate material sheet stacks. Once holding a stack, interactions are passed to the stack.
- Sheets in the new apparatus will always show as single sheets while held, regardless of the actual count.
- Removes the R-Glass tool from Engineer Cyborgs.
- Sabotage borgs have the same changes
## Why It's Good For The Game
Allows Engiborgs to repair or build with various materials, without (further) destroying the concept of material economy.

The R-Glass tool, in my experience, is generally unused, as the RCD can create R-Windows and that's the main use for the sheets. So the tool has been removed to keep the Engineer Borg Toolkit from exploding. The new tool can also pick up R-Glass, if a need for it arises.
## Changelog
:cl:
balance: Engineer Borgs now have a tool to manipulate material stacks (and also tile stacks). This replaces the R-Glass tool.
/:cl:
Pictured: Apparatus carrying titanium.
image

### CODE REVIEW

**Comment by YakumoChen** [comment]
borgs building r-walls sounds like its going to become an easy issue for balance.

**Comment by Helg2** [comment]
I mean its cool that you can construct something on yourself as engi-borg, but this is too abusable as malf.

**Comment by Sealed101** [comment]
> too abusable as malf.

good, malfies need more bullshit in their tactics

**Comment by Helg2** [comment]
> good, malfies need more bullshit in their tactics

I meant like if malf blows up all chem dispensers and build a giant square of reinforced walls it will take forever to get him.

**Comment by zxaber** [comment]
To an extent, you can already do this with regular walls. I don't think the downsides outweigh being able to properly repair the station.

**Comment by YakumoChen** [comment]
r-walls take 2-3 times the effort to get through, and you can RCD them down

**Comment by Helg2** [comment]
> To an extent, you can already do this with regular walls. I don't think the downsides outweigh being able to properly repair the station.

You can disassemble simple walls with just welder. For the reinforced ones you need a full toolbox and a plenty of time. And if malf gone delta, good luck disassembling 5 reinforced walls without any explosives.

**Comment by zxaber** [comment]
Airlocks take a similar amount of time to break and can be spammed using an RCD. And Malf AI has a power solely dedicated to blowing up RCDs to prevent easy removal. I don't expect mass R-wall to be the choice of many Malf players, given both construction time and also that their borg would be constantly retrieving plasteel stacks.

**Comment by Helg2** [comment]
> Airlocks take a similar amount of time to break and can be spammed using an RCD. And Malf AI has a power solely dedicated to blowing up RCDs to prevent easy removal. I don't expect mass R-wall to be the choice of many Malf players, given both construction time and also that their borg would be constantly retrieving plasteel stacks.

You can broke an airlock with a spear. And you can't do it with a wall. 

And there is always engineers with their bolter wrenches and skillchips of knowing wires. Is there any wires in reinforced wall?

**Comment by zxaber** [comment]
Honestly it ends up being a pain for projects. Getting an engineer to replace the two plasma glass sections in the emitter room can be like pulling teeth if I deconstructed them to move the airlocks. Nevermind for wall repairs on the other side of the station where you can't just finish a half-destroyed wall because it has plasteel wires on it.

**Comment by YakumoChen** [comment]
I was certain at one point you could "finish" r-girders with metal to make it a regular wall but you could also just remove the wires

You can move windows without turning them into base mats too

**Comment by YakumoChen** [comment]
Would be an interesting idea if using a screwdriver on a glass material let you make windows as a workaround

**Comment by Zergspower** [comment]
I'm not seeing the bigger issue given basically multiple codebases and plenty of downstream ones already give a plasteel stack or more and those concerns dont seem to be there relating to r-walls? If anything this solves the 'infinite glass and iron' problem by making it use station resources now which to me is a net win

**Comment by YakumoChen** [comment]
> I'm not seeing the bigger issue given basically multiple codebases and plenty of downstream ones already give a plasteel stack or more and those concerns dont seem to be there relating to r-walls? If anything this solves the 'infinite glass and iron' problem by making it use station resources now which to me is a net win - plonk it down with adv engineering

Skyrat malf is mollycoddled to hell and back, there is no way anyone would think to powergame that hard, so this is absolutely not a point to make. It's not a data point of value.

Not mentioning anyway that the skyrat plateel is like an unrenewable stack of 10.

**Comment by BrandonSmith6** [comment]
Borg's are cool don't get me wrong but I don't think giving them the ability to easily make reinforced stuff is maybe too much.

**Comment by RikuTheKiller** [comment]
R-walls? Good, finally. Manual construction is MUCH slower than RCD anyways.

**Comment by Zergspower** [comment]
> 

what? it's renewable jsut drag a stack behind you. It just doesn't recharge in the borg charger. This is something i've been planning to do anyway to get rid of the build in ones downstream for a blanket material grabber. You're making this a mountain  when it's literally not

**Comment by Zergspower** [comment]
> > I'm not seeing the bigger issue given basically multiple codebases and plenty of downstream ones already give a plasteel stack or more and those concerns dont seem to be there relating to r-walls? If anything this solves the 'infinite glass and iron' problem by making it use station resources now which to me is a net win - plonk it down with adv engineering
> 
> Skyrat malf is mollycoddled to hell and back, there is no way anyone would think to powergame that hard, so this is absolutely not a point to make. It's not a data point of value.
> 
> Not mentioning anyway that the skyrat plateel is like an unrenewable stack of 10 and requires a researched module so it makes little difference.

what? it's renewable jsut drag a stack behind you. It just doesn't recharge in the borg charger. This is something i've been planning to do anyway to get rid of the build in ones downstream for a blanket material grabber. You're making this a mountain when it's literally not

**Review by Kylerace — COMMENTED** [review]
code looks good, tentatively for this on the basis that it removes a dumb borgism (electricity for glass sheets) and forces them to use the same resources that the station has. and that while rwalls cannot be meleed or shot down, they can be blown up (like airlocks already) or thermited (unlike airlocks). if it becomes a serious issue we can always change it so that they cant do rwalls

**Comment by YakumoChen** [comment]
Plasteel security glass airlock walls with plasteel wall inner core might become the new meta, especially when the quickest way through walls is beakers from chemistry (I blew up chemistry 5 minutes ago)

**Comment by Zergspower** [comment]
RCDing 50 doors and electrifying them is a far larger deterrent tbh, far easier and way faster to do

**Comment by Helg2** [comment]
You needed to merge it on april fools and just see the results.

**Comment by Helg2** [comment]
> RCDing 50 doors and electrifying them is a far larger deterrent tbh, far easier and way faster to do

Bolter wrench and insulated gloves.

**Comment by YakumoChen** [comment]
> RCDing 50 doors and electrifying them is a far larger deterrent tbh, far easier and way faster to do

And you've doubled the time it takes to go through a door now with plasteel in the panels, so good job.

File: code/game/objects/items/robot/items/storage.dm [hunk 1]
```diff
@@ -230,6 +230,40 @@
...
+	. = ..()
+	var/mutable_appearance/arm = mutable_appearance(icon, "borg_stack_apparatus_arm1")
+	if(stored)
+		stored.pixel_x = 0
+		stored.pixel_y = 0
+		arm.icon_state = "borg_stack_apparatus_arm2"
+		var/mutable_appearance/stored_copy = new /mutable_appearance(stored)
+		var/underscore = findtext(stored_copy.icon_state, "_")
+		if(underscore)
+			stored_copy.icon_state = copytext(stored_copy.icon_state, 1, underscore) //how we use the icon_state of single sheets, even with full stacks
```
Comment (Fikou) [inline code/game/objects/items/robot/items/storage.dm:254]: just use initial() this is silly

Comment (zxaber) [reply code/game/objects/items/robot/items/storage.dm]: Alright, sure, that's probably better.

**Comment by zxaber** [comment]
image

The apparatus now supports stacks of floor tiles, too.

**Comment by Ryusenshu** [comment]
After playing a round with it i, kinda take back my criticism
Felt pretty fun to use and works pretty well into that tool set

There is still the issue about "balance" with the R-walls, still on the fence on that one

**Comment by dragomagol** [comment]
We'll see how the balance goes. In the meantime, I think we'll be fine.

**Pull request merged**